flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/TwoPhaseCommitSinkFunction.java

```diff
@@ -383,12 +404,14 @@ public void initializeState(FunctionInitializationContext context) throws Except
 
                 {
                     TXN transaction = operatorState.getPendingTransaction().handle;
-                    recoverAndAbort(transaction);
-                    handledTransactions.add(transaction);
-                    LOG.info(
-                            "{} aborted recovered transaction {}",
-                            name(),
-                            operatorState.getPendingTransaction());
+                    if (transaction != null) {
```

Comment:
The same question about test coverage here?

Reply:
This technically is exactly a re-write. The only reason adding a null check check is because we now my have empty holder that contains null TXN.

However, it is not possible to get the pendingTransaction as a empty one; as you can see from the adding logic, empty transaction will not be added to pending.

I